### PR TITLE
update github actions msrv to 1.57 so that base64 will build

### PR DESCRIPTION
**Pull request by jbr**
#1148 made CI red on main. This is the minimum change from #1140 that is required to make CI green

### CODE REVIEW

**Comment by djc** [comment]
#1148 is closed, so that was not the culprit -- rather, we're now pulling in base64 0.21 through rustls-pemfile.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
           - stable
           - beta
           - nightly
-          - 1.56 # MSRV
+          - 1.57 # MSRV
```
Comment (djc) [inline .github/workflows/build.yml:23]: Did you check for other mentions of the MSRV? We should at least update the `rust-version` in the Cargo metadata.

Comment (jbr) [reply .github/workflows/build.yml:23]: I wasn't sure how much of #1140 we wanted to duplicate. Will add rust-version and the readme comment as well.

File: rustls/Cargo.toml [hunk 2]
```diff
@@ -21,6 +21,7 @@ log = { version = "0.4.4", optional = true }
 ring = "0.16.20"
 sct = "0.7.0"
 webpki = { version = "0.22.0", features = ["alloc", "std"] }
+zeroize = { version = "1.5.7", features = ["derive"] }
```
Comment (djc) [inline rustls/Cargo.toml:24]: This seems unrelated?

**Review by djc — APPROVED** [review]

File: rustls/Cargo.toml (hunk 2 above)
Comment (jbr) [reply rustls/Cargo.toml:24]: entirely unrelated! 😬

**Review by djc — APPROVED** [review]
